libyul/AssemblyStack.cpp

```diff
@@ -249,21 +249,38 @@ pair<MachineAssemblyObject, MachineAssemblyObject> AssemblyStack::assembleAndGue
 	);
 
 	MachineAssemblyObject runtimeObject;
-	// Heuristic: If there is a single sub-assembly, this is likely the runtime object.
-	if (assembly.numSubs() == 1)
+
+	auto initRuntimeObject = [&](size_t i)
```

Comment:
Using a lambda function here feels kind of backwards. What about using an `optional<size_t>` for the sub index instead?